src/DssSpell.t.sol

```diff
@@ -1600,27 +1856,148 @@ contract DssSpellTest is DSTest, DSMath {
+        assertTrue(!oracle.good(ilk));
     }
+
+    function testSpellIsCast_RWA002_INTEGRATION_TELL_CURE_GOOD() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        tellSpell = new TellSpell();
+        vote(address(tellSpell));
+
+        tellSpell.schedule();
+
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        tellSpell.cast();
+        (, , uint48 tau,) = oracle.ilks(ilk);
+        if (tau == 0) {
+            assertTrue(!oracle.good(ilk));
+        } else {
+            assertTrue(oracle.good(ilk));
+        }
+        hevm.warp(block.timestamp + 600);
+        assertTrue(!oracle.good(ilk));
+
+        cureSpell = new CureSpell();
+        vote(address(cureSpell));
+
+        cureSpell.schedule();
+        castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        cureSpell.cast();
+        assertTrue(oracle.good(ilk));
+        (,,, uint48 toc) = oracle.ilks(ilk);
+        assertEq(uint256(toc), 0);
+    }
+
+    function testFailSpellIsCast_RWA002_INTEGRATION_CURE() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        cureSpell = new CureSpell();
+        vote(address(cureSpell));
+
+        cureSpell.schedule();
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        cureSpell.cast();
+    }
+
+    function testSpellIsCast_RWA002_INTEGRATION_TELL_CULL() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+        (, , uint48 tau,) = oracle.ilks(ilk);
+        assertTrue(oracle.good(ilk));
+
+        tellSpell = new TellSpell();
+        vote(address(tellSpell));
+
+        tellSpell.schedule();
+
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        tellSpell.cast();
+        if (tau == 0) {
+            assertTrue(!oracle.good(ilk));
+        } else {
+            assertTrue(oracle.good(ilk));
+        }
+        hevm.warp(block.timestamp + 600);
+        assertTrue(!oracle.good(ilk));
+
+        cullSpell = new CullSpell();
+        vote(address(cullSpell));
+
+        cullSpell.schedule();
+        castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        cullSpell.cast();
+        assertTrue(!oracle.good(ilk));
+        (, address pip,,) = oracle.ilks(ilk);
+        assertEq(DSValueAbstract(pip).read(), bytes32(0));
+    }
+
+    function testSpellIsCast_RWA002_END() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        endSpell = new EndSpell();
+        vote(address(endSpell));
+
+        endSpell.schedule();
+
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        endSpell.cast();
```

Comment:
Should there be more asserts / operations here? This is seems to just trigger the `end` and nothing else.

Reply:
I added an assert that end.live() eq 0, but everything else handled after that should be covered by dss testing.